app/DAO/BakeryDatabase.scala

```diff
@@ -68,6 +83,14 @@ class BakeryDatabase @Inject() (
     }
   }
 
+  def getProductByIdDoobie(id: String): Future[Option[Product]] = {
+    val query =
+      sql"""SELECT * FROM product where id::text = $id""".query[Product]
+    val action = query.option
+    println(action.toString())
+    action.transact(xa).unsafeToFuture()
```

Comment:
```suggestion
    val query =
      sql"""SELECT * FROM product where id::text = $id"""
      .query[Product]
      .option
    println(query.toString())
    query
      .transact(xa)
      .unsafeToFuture()
```

you can link the option part to the original query

Reply:
reformatted